charts/opentelemetry-operator/templates/deployment.yaml

```diff
@@ -89,6 +86,12 @@ spec:
               value: {{ $value | quote -}}
             {{- end }}
           {{- end }}
+          {{- if .Values.manager.createRbacPermissions}}
```

Comment:
We need to make sure the `env` header is present when .Values.manager.createRbacPermissions is true but .Values.manager.env has no values

Reply:
+1 another option would be to append this env var to the `.Values.manager.env` list if its not present already